Mathlib/RingTheory/LinearDisjoint.lean

```diff
@@ -412,6 +471,132 @@ variable {A B}
 
 variable (H : A.LinearDisjoint B)
 
+variable (R) in
+/-- If `A` and `B` are flat algebras over `R`, such that the algebra maps are injective, then
+their images in `A ⊗[R] B` are linearly disjoint. Note: they inject into `A ⊗[R] B`,
+see `Algebra.TensorProduct.includeLeft_injective` and
+`Algebra.TensorProduct.includeRight_injective`. -/
+theorem include_range_of_injective (A : Type v) [CommRing A] (B : Type w) [CommRing B]
+    [Algebra R A] [Algebra R B] [Module.Flat R A] [Module.Flat R B]
+    (ha : Function.Injective (algebraMap R A)) (hb : Function.Injective (algebraMap R B)) :
+    (Algebra.TensorProduct.includeLeft : A →ₐ[R] A ⊗[R] B).range.LinearDisjoint
+      (Algebra.TensorProduct.includeRight : B →ₐ[R] A ⊗[R] B).range := by
```

Comment:
Could you give some details? If `R` is not a field, then these two maps are not necessarily injective, and these two modules are not necessarily flat. Without these conditions I don't know how to prove the result.

Reply:
I agree with you that we don't need AlgHom.LinearDisjoint, only the LinearMap version. I'm thinking of the definitions:
```
import Mathlib

variable {R A B C} [CommSemiring R] [AddCommMonoid A] [AddCommMonoid B]
  [Module R A] [Module R B]
  
section

variable [NonUnitalNonAssocSemiring C] [Module R C]
  [SMulCommClass R C C] [IsScalarTower R C C]

def LinearMap.LinearDisjoint (f : A →ₗ[R] C) (g : B →ₗ[R] C) : Prop :=
  Function.Injective ((LinearMap.mul R C).compl₁₂ f g)
  
def Submodule.LinearDisjoint' (M N : Submodule R C) : Prop :=
  M.subtype.LinearDisjoint N.subtype

end

def Subalgebra.LinearDisjoint' [Semiring C] [Algebra R C] (M N : Subalgebra R C) :
  Prop := M.val.toLinearMap.LinearDisjoint N.val.toLinearMap

def IntermediateField.LinearDisjoint' {R C} [Field R] [Field C] [Algebra R C]
    (M N : IntermediateField R C) : Prop :=
  M.val.toLinearMap.LinearDisjoint N.val.toLinearMap
```
Instead of talking about the **range**s of Algebra.TensorProduct.includeLeft/Right are linearly disjoint, we would talk about the two linear maps being linearly disjoint, and in this case it's always true because it's saying the identity map is injective.

Reply:
Interesting. Let me see later if `Submodule.LinearDisjoint` should be refactored to your definition, in another PR. Currently let me think if `include_range` is true without injectivity and flatness assumptions.

Reply:
Oh yeah, I think I did convince myself that `include_range` is unconditionally true, because you can construct maps between `im(A)⊗im(B)` and `A⊗B` both ways and show they compose to the identity both ways.

Reply:
> Oh yeah, I think I did convince myself that `include_range` is unconditionally true, because you can construct maps between `im(A)⊗im(B)` and `A⊗B` both ways and show they compose to the identity both ways.

Reminds me of [Submodule.lTensorOne](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/TensorProduct/Submodule.html#Submodule.lTensorOne). Maybe if `A,B` are `R`-algebras, `M,N` are `R`-submodules in `A,B` respectively, then `im(M)⊗im(N)` and `M⊗N` are isomorphic.

Reply:
OK `include_range` is done. I think no further definitions/maps should be added unless for a new PR.